spinalcordtoolbox/scripts/sct_maths.py

```diff
@@ -270,223 +315,465 @@ def __call__(self, _, namespace, values, option_string=None):
+        )
 
-# MAIN
-# ==========================================================================================
-def main(argv: Sequence[str]):
-    parser = get_parser()
-    arguments = parser.parse_args(argv)
-    verbose = arguments.v
-    set_loglevel(verbose=verbose)
 
-    fname_in = arguments.i
-    fname_out = arguments.o
-    output_type = arguments.type
+def apply_mul(current_value, *args):
+    """Implementation for -mul."""
+    # special case to operate on a 4D volume across the t axis
+    if isinstance(current_value, Image) and current_value.data.ndim == 4 and not args:
+        return Image(
+            np.prod(current_value.data, axis=3),
+            hdr=current_value.hdr,
+        )
 
-    # Open file(s)
-    im = Image(fname_in)
+    header, values = get_header_and_values(current_value, args)
+    if header is None:
+        return math.prod(values)
+    else:
+        return Image(
+            np.prod(values, axis=0),
+            hdr=header,
+        )
 
-    # Get "array operation" arguments passed to the parser
-    ordered_args = [(arg, value) for (arg, value) in arguments.ordered_args
-                    if arg not in ['i', 'o', 'type', 'shape', 'dim', 'v']]  # Include only array operations
-    if not ordered_args:
-        printv(parser.error('ERROR: you need to specify an operation to do on the input image'))
 
-    # Case 1: Compute similarity metrics
-    if any(getattr(arguments, similarity_arg) is not None for similarity_arg in ['mi', 'minorm', 'corr']):
-        if arguments.mi is not None:
-            # input 1 = from flag -i --> im
-            # input 2 = from flag -mi
-            im_2 = Image(arguments.mi)
-            compute_similarity(im, im_2, fname_out, metric='mi', metric_full='Mutual information', verbose=verbose)
+def apply_div(current_value, *args):
+    """Implementation for -div."""
+    header, [first, *rest] = get_header_and_values(current_value, args)
+    if header is None:
+        return first / math.prod(rest)
+    else:
+        return Image(
+            first / np.prod(rest, axis=0),
+            hdr=header,
+        )
 
-        elif arguments.minorm is not None:
-            im_2 = Image(arguments.minorm)
-            compute_similarity(im, im_2, fname_out, metric='minorm', metric_full='Normalized Mutual information',
-                               verbose=verbose)
 
-        else:
-            assert arguments.corr is not None
-            # input 1 = from flag -i --> im
-            # input 2 = from flag -mi
-            im_2 = Image(arguments.corr)
-            compute_similarity(im, im_2, fname_out, metric='corr', metric_full='Pearson correlation coefficient',
-                               verbose=verbose)
+def apply_mean(current_value, dim):
+    """Implementation for -mean."""
+    if current_value.data.ndim == 3 and dim == 't':
+        return current_value
+    return Image(
+        np.mean(current_value.data, ('x', 'y', 'z', 't').index(dim)),
```

Comment:
I think this operation makes no sense when `dim != 't'`? Something to discuss at the SCT meeting.

Reply:
That's my intuition as well, but definitely worth bringing up.

Reply:
I brought it up at the last SCT meeting, and I wish I had taken better notes. But I think we settled on:
- If we take a mean across one of the spatial dimensions (x, y, z), then we should make that axis survive as a length=1 dimension in the output shape. (Otherwise, the affine matrix becomes meaningless.)
- If we take a mean across the time dimension, then the result should be 3-dimensional.